File: src/main/java/com/addressbook/ignite/UserCreator.java
package com.addressbook.ignite;

import com.addressbook.model.User;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@SuppressWarnings("all")
public class UserCreator {
    public static void initUsers() {
        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
        List<String> adminRoles = new ArrayList<>();
        adminRoles.add("USER");
        adminRoles.add("ADMIN");
        User admin = new User("admin", encoder.encode("adminPass"), adminRoles);
        GridDAO.createOrUpdateUser(admin);
        User user = new User("user", encoder.encode("userPass"), Collections.singletonList("USER"));
        GridDAO.createOrUpdateUser(user);
    }
}

File: addressbook-main/src/main/kotlin/com/addressbook/ignite/UserCreator.kt
package com.addressbook.ignite

import com.addressbook.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.stereotype.Component
import java.util.*
import javax.annotation.PostConstruct

@Component
class UserCreator {

    @Autowired
    var igniteDao: IgniteClient? = null

    @PostConstruct
    fun initUsers() {
        val encoder = BCryptPasswordEncoder()
        val adminRoles = ArrayList<String>()
        adminRoles.add("USER")
        adminRoles.add("ADMIN")
        val admin = User("admin", encoder.encode("adminPass"), adminRoles)
        igniteDao?.createOrUpdateUser(admin)
        val user = User("user", encoder.encode("userPass"), Collections.singletonList("USER"))
        igniteDao?.createOrUpdateUser(user)
    }
}